actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -404,7 +404,7 @@ def check_box_tag(name, value = "1", checked = false, options = {})
       #   radio_button_tag 'color', "green", true, class: "color_input"
       #   # => <input checked="checked" class="color_input" id="color_green" name="color" type="radio" value="green" />
       def radio_button_tag(name, value, checked = false, options = {})
-        html_options = { "type" => "radio", "name" => name, "id" => "#{sanitize_to_id(name)}_#{sanitize_to_id(value)}", "value" => value }.update(options.stringify_keys)
+        html_options = { "type" => "radio", "name" => name, "id" => "#{sanitize_to_id(name.tr("[", "_").delete("]"))}_#{value}", "value" => value }.update(options.stringify_keys)
```

Comment:
`.tr("[", "_").delete("]"))` shouldn't this go into `sanitize_to_id`?

Reply:
My intention here was to keep `radio_button_tag` id format as it was in case of grouped radios: group_option_id. Just in case is still relevant for someone as `sanitize_to_id` would render group[option]_value. I didn't want to affect [, ] characters that might be used by programmers in the rest of the helpers by applying it from `sanitize_to_id`.

I might as well leave it and let radio's id be like group[option]_value. Any thoughts on this?  : )